### PR TITLE
add write a new model file

### PR DESCRIPTION
**Pull request by chenyang23333**
Thank you for your contribution to the MindYOLO repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindyolo/blob/master/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: docs/zh/how_to_guides/write_a_new_model.md [hunk 1]
```diff
@@ -1,2 +1,205 @@
...
+        self.names = [str(i) for i in range(nc)]  # default names
+
+        self.reset_parameter()
+
+    def construct(self, x):
+        return self.model(x)
+  
+    def reset_parameter(self):  #定义了reset_parameter方法用于初始化模型的参数。
+
+        initialize_defult(self)
```
Comment (zhanghuiyao) [inline docs/zh/how_to_guides/write_a_new_model.md:113]: 1. reset_parameter 定义可以删掉，这个不是必须的；
2. 可以直接在 __init__ 中直接调用 initialize_defult 并描述原因：
```python
initialize_defult(self)  # 可选，你可能需要initialize_defult方法以获得和pytorch一样的conv2d、dense层的初始化方式；
```

File: docs/zh/how_to_guides/write_a_new_model.md [hunk 2]
```diff
@@ -1,2 +1,205 @@
...
+## 在模型函数中创建Mymodel模型实例
+mymodel函数被@register_model装饰器修饰，使其成为可调用的模型。在函数内通过MYmodel类创建了一个模型对象，并返回该对象作为模型。
+```python
+@register_model
+def mymodel(cfg, in_channels=3, num_classes=None, **kwargs) -> MYmodel:
+    """Get GoogLeNet model.
+    Refer to the base class `models.GoogLeNet` for more details."""
+    model = MYmodel(cfg=cfg, in_channels=in_channels, num_classes=num_classes, **kwargs)
+    return model
+```
```
Comment (zhanghuiyao) [inline docs/zh/how_to_guides/write_a_new_model.md:125]: 这段可以删掉，和下面一样

File: docs/zh/how_to_guides/write_a_new_model.md [hunk 3]
```diff
@@ -1,2 +1,205 @@
...
+        return x
+```
+### 创建模型实例
+通过实例化MyModel类，创建一个模型实例model，后续可以使用该实例进行模型的训练和推理。
+```python
+
+model = MyModel()
+
+```
+
```
Comment (zhanghuiyao) [inline docs/zh/how_to_guides/write_a_new_model.md:183]: ### 注册模型 (可选)

如果需要使用mindyolo接口初始化自定义的模型，那么需要先对模型进行 `注册` 和 `导入`

 1. 模型注册  # 注册后的模型可以通过 create_model 接口以模型名的方式进行访问；
...
2. 模型导入
...

File: docs/zh/how_to_guides/write_a_new_model.md [hunk 4]
```diff
@@ -1,2 +1,229 @@
...
+        self.stride_max = int(max(self.cfg.stride))
+        ch, nc = in_channels, num_classes
+
+        self.nc = nc  # override yaml value
+        self.model = build_model_from_cfg(model_cfg=cfg, in_channels=ch, num_classes=nc, sync_bn=sync_bn)
+        self.names = [str(i) for i in range(nc)]  # default names
+        
+        initialize_defult()  # 可选，你可能需要initialize_defult方法以获得和pytorch一样的conv2d、dense层的初始化方式；
+
+        self.reset_parameter()
```
[the remark this answers is not in the document]
Comment (chenyang23333) [reply docs/zh/how_to_guides/write_a_new_model.md:157]: fixed

File: docs/zh/how_to_guides/write_a_new_model.md [hunk 5]
```diff
@@ -1,2 +1,229 @@
...
+```
+### 创建模型
+在`MindSpore`中，模型的类继承于`nn.Cell`，一般来说需要重载以下两个函数：
+
+- 在`__init__`函数中，应当定义模型中需要用到的module层。
+- 在`construct`函数中定义模型前向逻辑。 <br>
+
+```python
+
+
```
[the remark this answers is not in the document]
Comment (chenyang23333) [reply docs/zh/how_to_guides/write_a_new_model.md:141]: fixed

File: docs/zh/how_to_guides/write_a_new_model.md [hunk 6]
```diff
@@ -1,2 +1,229 @@
...
+    [[-1, 1, ConvNormAct, [32, 3, 1]],  # 0
+     [-1, 1, ConvNormAct, [64, 3, 2]],  # 1-P1/2
+     [-1, 1, Bottleneck, [64]],
+     [-1, 1, ConvNormAct, [128, 3, 2]],  # 3-P2/4
+     [-1, 2, Bottleneck, [128]],
+     [-1, 1, ConvNormAct, [256, 3, 2]],  # 5-P3/8
+     [-1, 8, Bottleneck, [256]],
+      ]
+  
+  head: #head部分的配置
```
[the remark this answers is not in the document]
Comment (chenyang23333) [reply docs/zh/how_to_guides/write_a_new_model.md:97]: fixed

File: docs/zh/how_to_guides/write_a_new_model.md [hunk 7]
```diff
@@ -1,2 +1,223 @@
...
-即将到来
+本文档提供MindYOLO编写自定义模型的教程。<br>
+分为三个部分：
+- 模型定义：我们可以直接定义一个网络，也可以使用yaml文件方式定义一个网络。
+- 注册模型：可选，注册之后可以在create_model接口中使用文件名创建自定义的模型
+- 验证: 验证模型是否可运行
+
+## 模型定义
+### 1.直接使用python代码来编写网络
+### 模块导入
```
Comment (CaitinZhao) [inline docs/zh/how_to_guides/write_a_new_model.md:10]: #### 模块导入

File: docs/zh/how_to_guides/write_a_new_model.md [hunk 8]
```diff
@@ -1,2 +1,223 @@
...
+导入MindSpore框架中的nn模块和ops模块，用于定义神经网络的组件和操作。
+```python
+import mindspore.nn as nn
+import mindspore.ops.operations as ops
+```
+
+### 创建模型
+定义了一个继承自nn.Cell的模型类MyModel。在构造函数__init__中，定义模型的各个组件：
```
Comment (CaitinZhao) [inline docs/zh/how_to_guides/write_a_new_model.md:18]: #### 创建模型

File: docs/zh/how_to_guides/write_a_new_model.md [hunk 9]
```diff
@@ -1,2 +1,223 @@
...
+        x = self.relu(x)
+        x = self.maxpool(x)
+        x = self.conv2(x)
+        x = self.relu(x)
+        x = self.maxpool(x)
+        x = x.view(x.shape[0], -1)
+        x = self.fc(x)
+        return x
+```
+### 创建模型实例
```
Comment (CaitinZhao) [inline docs/zh/how_to_guides/write_a_new_model.md:47]: #### 创建模型实例

File: docs/zh/how_to_guides/write_a_new_model.md [hunk 10]
```diff
@@ -1,2 +1,223 @@
...
+通过实例化MyModel类，创建一个模型实例model，后续可以使用该实例进行模型的训练和推理。
+```python
+
+model = MyModel()
+
+```
+
+## 2.使用yaml文件编写网络
```
Comment (CaitinZhao) [inline docs/zh/how_to_guides/write_a_new_model.md:55]: ### 2.使用yaml文件编写网络

File: docs/zh/how_to_guides/write_a_new_model.md [hunk 11]
```diff
@@ -1,2 +1,223 @@
...
+通常需要以下三个步骤
+- 新建一个mymodel.yaml文件
+- 新建对应的mymodel.py文件 
+- 在mindyolo/models/_init_.py文件中引入该模型
+
+### 以下是编写mymodel.yaml文件的详细指导
```
Comment (CaitinZhao) [inline docs/zh/how_to_guides/write_a_new_model.md:61]: 以下是编写mymodel.yaml文件的详细指导：

Comment (chenyang23333) [reply docs/zh/how_to_guides/write_a_new_model.md]: fixed

File: docs/zh/how_to_guides/write_a_new_model.md (hunk 8 above)
Comment (chenyang23333) [reply docs/zh/how_to_guides/write_a_new_model.md:20]: fixed

**Review by zhanghuiyao — APPROVED** [review]

**Review by CaitinZhao — APPROVED** [review]

**Pull request merged**